## **BIOQUÍMICA E INGENIERÍA DE PROTEÍNAS**


# ESTABILIZACIÓN TÉRMICA DE COLD SHOCK PROTEINS

Javier Amaya Jiménez, Zeneida Carneros Gómez, Eva Pérez Abad y Lorena Romero Hernández.

### 1. INTRODUCCIÓN



La proteína de choque térmico frío de Bacillus subtilis (Bs‑CspB) constituye uno de los modelos más ampliamente estudiados para comprender los principios del plegamiento de proteínas, la estabilidad térmica y las interacciones proteína‑ácido nucleico (Schindelin et al., 1993).

Se trata de una proteína de pequeño tamaño, compuesta por 67 aminoácidos, que presenta una estructura secundaria compacta de tipo barril beta. Este barril está compuesto por cinco láminas beta antiparalelas, organizadas formando una superficie hidrofóbica (Schindelin et al., 1993). Esta superficie es crucial para su interacción con ácidos nucleicos, en particular con moléculas de ARN o ADN de cadena sencilla (Max et al., 2007). Gracias a esta conformación, Bs‑CspB puede unirse con afinidad a ácidos nucleicos de cadena sencilla (ssDNA y ssRNA), llevando a cabo un papel esencial en la estabilización de estas moléculas a bajas temperaturas y evitando la formación de estructuras complejas que pudieran interferir con los procesos celulares (Martin et al., 2002).

Funcionalmente, nuestra proteína de estudio actúa como un factor de transcripción y como una chaperona de ARN, participando en la regulación global de la expresión génica (Faßhauer et al., 2021). Su acción contribuye a mantener la eficiencia de la maquinaria genética bajo condiciones que no son favorables, lo que facilita la adaptación de B. subtilis en entornos donde la temperatura desciende bruscamente (Brandi et al., 1991).

En este estudio, el objetivo es aumentar la estabilidad térmica de Bs‑CspB, aplicando un conjunto de mutaciones que incrementen su termoestabilidad sin comprometer su conformación global ni su función biológica. Más allá de buscar únicamente mutaciones puntuales, se busca realizar un diseño racional que cumpla unos criterios de estabilidad y conservación estructural óptimos para el funcionamiento de la proteína a temperaturas elevadas (Max et al., 2007; Martin et al., 2002).

### 2. DESCRIPCIÓN DEL DISEÑO

En primer lugar, se descarga la estructura PDB de Bs-CspB, con ID 1CSP de la base de datos usando el paquete "biopython".

In [ ]:
!pip install pyrosettacolabsetup
import pyrosettacolabsetup; pyrosettacolabsetup.install_pyrosetta()
import pyrosetta; pyrosetta.init()

┌───────────────────────────────────────────────────────────────────────────────┐
│                                  PyRosetta-4                                  │
│               Created in JHU by Sergey Lyskov and PyRosetta Team              │
│               (C) Copyright Rosetta Commons Member Institutions               │
│                                                                               │
│ NOTE: USE OF PyRosetta FOR COMMERCIAL PURPOSES REQUIRES PURCHASE OF A LICENSE │
│          See LICENSE.PyRosetta.md or email license@uw.edu for details         │
└───────────────────────────────────────────────────────────────────────────────┘
PyRosetta-4 2025 [Rosetta PyRosetta4.MinSizeRel.python312.ubuntu 2025.43+release.9fd5078c70dbd3d139526618074f384bba8cde37 2025-10-23T17:08:57] retrieved from: http://www.pyrosetta.org
core.init: Checking for fconfig files in pwd and ./rosetta/flags
core.init: Rosetta version: PyRosetta4.MinSizeRel.python312.ubuntu r414 2025.43+release.9fd5078

In [ ]:
!pip install biopython

In [ ]:
import os
from pyrosetta import pose_from_pdb
from Bio.PDB import PDBList

pdb_id = "1CSP"
pdbl = PDBList()

# Se descarga el PDB y lo ponemos en el formato pdb
full_pdb_filepath = pdbl.retrieve_pdb_file(pdb_id, file_format='pdb')

# Carga el archivo pdb en una pose
full_pdb_pose = pose_from_pdb(full_pdb_filepath)

print(full_pdb_pose.sequence())

Structure exists: '/content/cs/pdb1csp.ent' 
core.import_pose.import_pose: File '/content/cs/pdb1csp.ent' automatically determined to be of type PDB from contents.
core.conformation.Conformation: [ WARNING ] missing heavyatom:  CG  on residue GLU 3
core.conformation.Conformation: [ WARNING ] missing heavyatom:  CD  on residue GLU 3
core.conformation.Conformation: [ WARNING ] missing heavyatom:  OE1 on residue GLU 3
core.conformation.Conformation: [ WARNING ] missing heavyatom:  OE2 on residue GLU 3
core.conformation.Conformation: [ WARNING ] missing heavyatom:  CG  on residue GLU 21
core.conformation.Conformation: [ WARNING ] missing heavyatom:  CD  on residue GLU 21
core.conformation.Conformation: [ WARNING ] missing heavyatom:  OE1 on residue GLU 21
core.conformation.Conformation: [ WARNING ] missing heavyatom:  OE2 on residue GLU 21
core.conformation.Conformation: [ WARNING ] missing heavyatom:  CG  on residue GLU 36
core.conformation.Conformation: [ WARNING ] missing heavyatom:  CD

Se comprueba que la secuencia de Bs-CspB se ha descargado correctamente:


In [ ]:
print(full_pdb_pose.sequence())

MLEGKVKWFNSEKGFGFIEVEGQDDVFVHFSAIQGEGFKTLEEGQAVSFEIVEGNRGPQAANVTKEA


A partir de aquí, se nombra a esta secuencia como "seq_problema" porque será a la que se aplique las mutaciones para mejorar la termoestabilidad:

In [ ]:
seq_problema = full_pdb_pose.sequence()

In [ ]:
print(seq_problema)

MLEGKVKWFNSEKGFGFIEVEGQDDVFVHFSAIQGEGFKTLEEGQAVSFEIVEGNRGPQAANVTKEA


Se observa la estructura de Bs-CspB, empleando "py3Dmol":

In [ ]:
!pip install py3Dmol

In [ ]:
import py3Dmol
# Descarga el PDB desde RCSB
pdb_id = "1CSP"
pdbl = PDBList()
full_pdb_filepath = pdbl.retrieve_pdb_file(pdb_id, file_format='pdb')

# Carga la proteína en PyRosetta
pose = pose_from_pdb(full_pdb_filepath)

# Guarda el pose en un archivo PDB temporal
pose.dump_pdb("1CSP_clean.pdb")

# Lee el PDB y mostrar en py3Dmol
with open("1CSP_clean.pdb") as f:
    pdb_data = f.read()

view = py3Dmol.view(width=600, height=400)
view.addModel(pdb_data, "pdb")
view.setStyle({"cartoon": {"color": "spectrum"}})  # cartoon con gradiente de colores
view.zoomTo()
view.show()

Structure exists: '/content/cs/pdb1csp.ent' 
core.import_pose.import_pose: File '/content/cs/pdb1csp.ent' automatically determined to be of type PDB from contents.
core.conformation.Conformation: [ WARNING ] missing heavyatom:  CG  on residue GLU 3
core.conformation.Conformation: [ WARNING ] missing heavyatom:  CD  on residue GLU 3
core.conformation.Conformation: [ WARNING ] missing heavyatom:  OE1 on residue GLU 3
core.conformation.Conformation: [ WARNING ] missing heavyatom:  OE2 on residue GLU 3
core.conformation.Conformation: [ WARNING ] missing heavyatom:  CG  on residue GLU 21
core.conformation.Conformation: [ WARNING ] missing heavyatom:  CD  on residue GLU 21
core.conformation.Conformation: [ WARNING ] missing heavyatom:  OE1 on residue GLU 21
core.conformation.Conformation: [ WARNING ] missing heavyatom:  OE2 on residue GLU 21
core.conformation.Conformation: [ WARNING ] missing heavyatom:  CG  on residue GLU 36
core.conformation.Conformation: [ WARNING ] missing heavyatom:  CD

3Dmol.js failed to load for some reason. Please check your browser console for error messages.

Se observa claramente la estructura de barril beta que se indicaba en la introducción.

**Flujo de trabajo:**

**Paso 1: Creación de la estructura relajada**

Se aplica el protocolo "FastRelax" a la estructura 1CSP, usando un *resfile* que contiene solo "NATRO". De esta forma, se genera una estructura que representa el estado de mínima energía, llamada "pose_relax":

In [ ]:
# nodesign resfile
resfile_content = """
NATRO
start


"""

# Ruta en el que se guardará el resfile
output_resfile_path = "/content/nodesign.resfile"

# Contenido del archivo
with open(output_resfile_path, "w") as f:
    f.write(resfile_content)

print(f"Resfile saved to: {output_resfile_path}")

Resfile saved to: /content/nodesign.resfile


In [ ]:
# new design resfile
nodesign_resfile = "/content/nodesign.resfile"

pose_relax = full_pdb_pose.clone()

tf = pyrosetta.rosetta.core.pack.task.TaskFactory()
tf.push_back(pyrosetta.rosetta.core.pack.task.operation.InitializeFromCommandline())
tf.push_back(pyrosetta.rosetta.core.pack.task.operation.IncludeCurrent())


# Se incluye el resfile
tf.push_back(pyrosetta.rosetta.core.pack.task.operation.ReadResfile(nodesign_resfile))

packer_task = tf.create_task_and_apply_taskoperations(pose_relax)
print(packer_task)


mm = pyrosetta.rosetta.core.kinematics.MoveMap()
mm.set_bb(True)
mm.set_chi(True)
mm.set_jump(True)

scorefxn = pyrosetta.create_score_function("ref2015_cart.wts")
rel_design = pyrosetta.rosetta.protocols.relax.FastRelax(scorefxn_in=scorefxn, standard_repeats=1, script_file="MonomerDesign2019")
rel_design.cartesian(True)
rel_design.set_task_factory(tf)
rel_design.set_movemap(mm)
rel_design.minimize_bond_angles(True)
rel_design.minimize_bond_lengths(True)

%time rel_design.apply(pose_relax)

core.pack.task: Packer task: initialize from command line()
#Packer_Task

Threads to request: ALL AVAILABLE

resid	pack?	design?	allowed_aas
1	FALSE	FALSE	
2	FALSE	FALSE	
3	FALSE	FALSE	
4	FALSE	FALSE	
5	FALSE	FALSE	
6	FALSE	FALSE	
7	FALSE	FALSE	
8	FALSE	FALSE	
9	FALSE	FALSE	
10	FALSE	FALSE	
11	FALSE	FALSE	
12	FALSE	FALSE	
13	FALSE	FALSE	
14	FALSE	FALSE	
15	FALSE	FALSE	
16	FALSE	FALSE	
17	FALSE	FALSE	
18	FALSE	FALSE	
19	FALSE	FALSE	
20	FALSE	FALSE	
21	FALSE	FALSE	
22	FALSE	FALSE	
23	FALSE	FALSE	
24	FALSE	FALSE	
25	FALSE	FALSE	
26	FALSE	FALSE	
27	FALSE	FALSE	
28	FALSE	FALSE	
29	FALSE	FALSE	
30	FALSE	FALSE	
31	FALSE	FALSE	
32	FALSE	FALSE	
33	FALSE	FALSE	
34	FALSE	FALSE	
35	FALSE	FALSE	
36	FALSE	FALSE	
37	FALSE	FALSE	
38	FALSE	FALSE	
39	FALSE	FALSE	
40	FALSE	FALSE	
41	FALSE	FALSE	
42	FALSE	FALSE	
43	FALSE	FALSE	
44	FALSE	FALSE	
45	FALSE	FALSE	
46	FALSE	FALSE	
47	FALSE	FALSE	
48	FALSE	FALSE	
49	FALSE	FALSE	
50	FALSE	FALSE	
51	FALSE	FALSE	
52	FALSE	FALSE	
53	FALSE	FALSE	
54	FALSE	FALSE	
55	F

La energía de la estructura relajada ("pose_relax") es la siguiente:

In [ ]:
energy_nodesign = scorefxn(pose_relax)
print(energy_nodesign)

-193.4128123861998


Aplicando "scorefxn.show", se observan los términos energéticos (fuerzas de repulsión y atracción de Van der Waals, puentes de hidrógeno, etc.)

In [ ]:
scorefxn.show(pose_relax)

core.scoring.ScoreFunction: 
------------------------------------------------------------
 Scores                       Weight   Raw Score Wghtd.Score
------------------------------------------------------------
 fa_atr                       1.000    -342.739    -342.739
 fa_rep                       0.550      63.244      34.784
 fa_sol                       1.000     175.128     175.128
 fa_intra_rep                 0.005     152.812       0.764
 fa_intra_sol_xover4          1.000      12.245      12.245
 lk_ball_wtd                  1.000      -7.463      -7.463
 fa_elec                      1.000    -102.993    -102.993
 hbond_sr_bb                  1.000      -7.075      -7.075
 hbond_lr_bb                  1.000     -35.995     -35.995
 hbond_bb_sc                  1.000      -6.899      -6.899
 hbond_sc                     1.000      -4.107      -4.107
 dslf_fa13                    1.250       0.000       0.000
 omega                        0.400       1.099       0.439
 fa_dun 

Para que el diseño se considere exitoso, el término energético repulsivo no debe aumentar más de **20 REU**.


In [ ]:
!install pyrosetta.rosetta.core.scoring
from pyrosetta.rosetta.core.scoring import ScoreType
fa_rep_nodesign = pose_relax.energies().total_energies()[ScoreType.fa_rep]
print(fa_rep_nodesign)

install: missing destination file operand after 'pyrosetta.rosetta.core.scoring'
Try 'install --help' for more information.
63.24379819406139


Para identificar los puentes salinos, se emplea una función que identifica los residuos cargados (lisina, arginina, aspartato o glutamato) y mide las distancias entre sus átomos cargados. Se aplica un límite para definir un puente salino, de manera que si la **distancia medida es menor de 4.0 Å**, lo incluye en una lista:

In [ ]:
from pyrosetta.rosetta.core.scoring.hbonds import HBondSet
from pyrosetta import Pose

def puentes_salinos(pose: Pose, sb_cutoff_dist: float) -> dict:

    sb_list = []

    # Átomos centrales de carga para la medición de distancia
    atomos_cargados = {'LYS': 'NZ', 'ARG': 'CZ', 'ASP': 'CG', 'GLU': 'CD'}
    res_positivos = {'LYS', 'ARG'}

    # Pre-filtrar solo residuos cargados para eficiencia
    res_cargados = [i for i in range(1, pose.total_residue() + 1) if pose.residue(i).name3() in atomos_cargados]

    # Compara cada residuo cargado con los demás
    for i in range(len(res_cargados)):
        for j in range(i + 1, len(res_cargados)):
            idx1, idx2 = res_cargados[i], res_cargados[j]
            res1, res2 = pose.residue(idx1), pose.residue(idx2)
            name1, name2 = res1.name3(), res2.name3()

            is_pos1 = name1 in res_positivos
            is_pos2 = name2 in res_positivos

            # Si son de cargas opuestas
            if is_pos1 != is_pos2:
                try:
                    # Mide la distancia entre los átomos de carga
                    atom1 = res1.atom(atomos_cargados[name1])
                    atom2 = res2.atom(atomos_cargados[name2])
                    distance = atom1.xyz().distance(atom2.xyz())

                    # Si la distancia es menor que el límite, es un puente salino
                    if distance < sb_cutoff_dist:
                        sb_list.append(f"{idx1:3d} {name1:3s} <-> {idx2:3d} {name2:3s}  ({distance:.2f} Å)")
                except:
                    # Ignora si el átomo no existe (salvaguarda)
                    pass

    return {
        "Lista de puentes salinos": sb_list,
        "Nº de puentes salinos": len(sb_list),
    }

In [ ]:
# Límite de distancia para Puentes Salinos (en Angstroms)
SB_CUTOFF_DISTANCE = 4.0

In [ ]:
# Calcular métricas de la línea base
baseline_metrics = puentes_salinos(pose_relax, SB_CUTOFF_DISTANCE)
baseline_sb_count = baseline_metrics['Nº de puentes salinos']

print(f"Línea Base WT: {baseline_sb_count} puentes salinos encontrados.")

Línea Base WT: 1 puentes salinos encontrados.


In [ ]:
# Print the list of salt bridges in the baseline metrics
print("Los puentes salinos se dan entre los aminoácidos:")
for sb in baseline_metrics["Lista de puentes salinos"]:
    print(sb)

Los puentes salinos se dan entre los aminoácidos:
  7 LYS <->  25 ASP  (3.66 Å)


Como podemos observar, la estructura original (1CSP) cuenta con un **puente salino entre la lisina 7 y el aspartato 25**, como se indica en el artículo de Mueller *et al.*, 2000.

Ahora, se observan los puentes de hidrógeno de la estructura original:

In [ ]:
from pyrosetta import Pose # Necesario para el type hinting

def puentes_hidrogeno(pose: Pose) -> dict:

    # Crea un objeto HBondSet (conjunto de puentes de H)
    hbond_set = HBondSet()

    hbond_set.setup_for_residue_pair_energies(pose, False, False)

    # Conteo total de puentes de hidrógeno
    total_hbonds = hbond_set.nhbonds()

    # Diferencia entre H-Bonds del esqueleto (BB) y el resto (SC)
    #    Itera sobre todos los puentes y suma 1 si donador y aceptor son átomos del esqueleto (backbone).
    bb_hbonds = sum(1 for h in hbond_set.hbonds()
                    if h.don_hatm_is_backbone() and h.acc_atm_is_backbone())

    # El resto son puentes de H que involucran cadenas laterales
    #    (SC-SC o SC-BB)
    sc_hbonds = total_hbonds - bb_hbonds

    # Devuelve los resultados
    return {
        "Nº H-Bonds (Total)": total_hbonds,
        "Nº H-Bonds (BB)": bb_hbonds,
        "Nº H-Bonds (SC)": sc_hbonds,
    }

In [ ]:
hbond_info = puentes_hidrogeno(pose_relax)

print("Análisis de Puentes de Hidrógeno:")
print()
print(f"  Total: {hbond_info['Nº H-Bonds (Total)']}")
print(f"  Backbone-Backbone: {hbond_info['Nº H-Bonds (BB)']}")
print(f"  Sidechain/Mixtos: {hbond_info['Nº H-Bonds (SC)']}")

Análisis de Puentes de Hidrógeno:

  Total: 44
  Backbone-Backbone: 36
  Sidechain/Mixtos: 8


Hay un total de **44 puentes de hidrógeno**, de los cuales 36 pertenecen al esqueleto (BB) y 8 son los puentes de hidrógeno formados con cadenas laterales.

Antes de continuar con los diseños, se elabora una lista donde se añadirá toda la información que se ha analizado para, finalmente, elaborar una tabla comparativa:

In [ ]:
lista_resultados = []
TM_WILD_TYPE = 49.0

Se define una función que va a recoger las métricas de evaluación de la proteína original:

In [ ]:

from pyrosetta.rosetta.core.scoring import CA_rmsd # Para calcular el RMSD (desviación estructural)
import pandas as pd # Para organizar y mostrar los resultados en tablas


def obtener_mutaciones_y_alineamiento(pose_wt: Pose, pose_diseno: Pose) -> (list, str):
    """
    Compara las secuencias de dos poses (WT y Diseño)
    Devuelve:
    1. Una lista de mutaciones aplicadas
    2. Una cadena de texto (string) con el alineamiento
    """
    seq_wt = pose_wt.sequence()
    seq_diseno = pose_diseno.sequence()
    lista_mutaciones = []
    simbolos_alineamiento = ""

    for i in range(len(seq_wt)):
        aa_wt = seq_wt[i]
        aa_diseno = seq_diseno[i]
        res_num = i + 1 # Numeración 1-based

        if aa_wt == aa_diseno:
            simbolos_alineamiento += "|"
        else:
            simbolos_alineamiento += " "
            lista_mutaciones.append(f"{aa_wt}{res_num}{aa_diseno}")

    # Formatea el string de alineamiento para que sea legible
    chunk_size = 60
    output_lines = ["Alineamiento de Secuencia:"]
    for i in range(0, len(seq_wt), chunk_size):
        start, end = i, i + chunk_size
        output_lines.append(f"  WT:     {start+1:<4} {seq_wt[start:end]}")
        output_lines.append(f"          {'':<4} {simbolos_alineamiento[start:end]}")
        output_lines.append(f"  Design: {start+1:<4} {seq_diseno[start:end]}\n")

    mut_str = f"Mutaciones aplicadas ({len(lista_mutaciones)}): {', '.join(lista_mutaciones) or 'Ninguna'}"
    output_lines.append(mut_str)

    return lista_mutaciones, "\n".join(output_lines)
def calcular_metricas_evaluacion(
    nombre_diseno: str,
    pose_wt: Pose,
    pose_diseno: Pose,
    tm_wt: float
) -> dict:
    """
    Calcula las métricas clave de evaluación (ΔΔG, RMSD, ΔTm)
    comparando una pose de diseño con la wild-type relajada.
    """

    # Qué mutaciones y el número de mutaciones
    lista_mutaciones, _ = obtener_mutaciones_y_alineamiento(pose_wt, pose_diseno)
    num_mutaciones = len(lista_mutaciones)

    # RMSD (cuánta estructura se conserva)
    rmsd_ca = CA_rmsd(pose_wt, pose_diseno)

    # Energía (ΔΔG)
    energia_wt = pose_wt.energies().total_energy()
    energia_diseno = pose_diseno.energies().total_energy()
    ddG = energia_diseno - energia_wt

    # Término repulsivo (Δfa_rep)
    fa_rep_wt = pose_wt.energies().total_energies()[ScoreType.fa_rep]
    fa_rep_diseno = pose_diseno.energies().total_energies()[ScoreType.fa_rep]
    delta_fa_rep = fa_rep_diseno - fa_rep_wt

    # Score de Estabilidad y Estimación de Tm

    # Factor de eficiencia basado en el número de mutaciones
    if num_mutaciones == 0:
        factor_eficiencia = 1.0
    elif num_mutaciones <= 3:
        factor_eficiencia = 1.0
    elif num_mutaciones <= 6:
        factor_eficiencia = 0.9
    else:
        factor_eficiencia = 0.8

    score_estabilidad = ddG * factor_eficiencia


    # SB_CUTOFF_DISTANCE está definida previamente
    metricas_sb_wt = puentes_salinos(pose_wt, SB_CUTOFF_DISTANCE)
    metricas_sb_diseno = puentes_salinos(pose_diseno, SB_CUTOFF_DISTANCE)
    metricas_hb_diseno = puentes_hidrogeno(pose_diseno)

    bonus_sb_aplicado = "No"

    # Comprobamos si el diseño tiene MÁS puentes salinos que el WT
    if metricas_sb_diseno["Nº de puentes salinos"] > metricas_sb_wt["Nº de puentes salinos"]:
        score_estabilidad *= 1.05 # Bonus del 5%
        bonus_sb_aplicado = "Sí"

    # Estimación de Tm
    FACTOR_ESTIMACION_TM = -7.1
    delta_tm_estimado = FACTOR_ESTIMACION_TM * ddG
    tm_diseno = tm_wt + delta_tm_estimado

    # Guarda toda esta información en "metricas"
    metricas = {
        "Diseño": nombre_diseno,
        "Nº Mutaciones": num_mutaciones,
        "RMSD (Å)": rmsd_ca,
        "Score Total (REU)": energia_diseno,
        "ΔΔG (REU)": ddG,
        "Δfa_rep": delta_fa_rep,
        "Score Estabilidad": score_estabilidad,
        "ΔTm Estimado (°C)": delta_tm_estimado,
        "Tm Diseño (°C)": tm_diseno,
        "Bonus SB": bonus_sb_aplicado,
        # Añadimos los datos de TUS funciones:
        "Dis_Nº Puentes Salinos": metricas_sb_diseno["Nº de puentes salinos"],
        "Dis_Nº H-Bonds (Total)": metricas_hb_diseno["Nº H-Bonds (Total)"],
        "Dis_Nº H-Bonds (BB)": metricas_hb_diseno["Nº H-Bonds (BB)"],
        "Dis_Nº H-Bonds (SC)": metricas_hb_diseno["Nº H-Bonds (SC)"]
    }

    return metricas

In [ ]:
print("ANÁLISIS DE LA ESTRUCTURA ORIGINAL (WT) RELAJADA")
print()
metricas_wt = calcular_metricas_evaluacion(
    nombre_diseno="Wild-Type",
    pose_wt=pose_relax,
    pose_diseno=pose_relax,
    tm_wt=TM_WILD_TYPE
)
lista_resultados.append(metricas_wt)
print(f"Energía WT: {metricas_wt['Score Total (REU)']:.2f} REU")
print()
print(f"Puentes Salinos WT: {metricas_wt['Dis_Nº Puentes Salinos']}")

ANÁLISIS DE LA ESTRUCTURA ORIGINAL (WT) RELAJADA

Energía WT: -193.41 REU

Puentes Salinos WT: 1


**Paso 2: Introducción de las mutaciones**

A continuación, se definen un conjunto de mutaciones en el *resfile* para estudiar sus cambios en la estructura. Tras analizar numerosos conjuntos de mutaciones, se exponen y comparan tres diseños, siendo el último nuestro resultado final, a fin de explorar la evolución de nuestro trabajo y enfoque racional.

Conjunto de mutaciones que se comprobaron:

- Q34V, ΔΔG = -0.2851956986792459

- E42P, ΔΔG = -1.4539471763817176

- Q34V y E42P, ΔΔG = -2.2088301503052037

Mutaciones recogidas en Mueller *et al*. (2000) y sus combinaciones:

-	D24S, ΔΔG = -2.5358120582151003

-	E3R y E66L, ΔΔG = -5.3640488031972495

-	D24S, E3R y E66L, ΔΔG = -5.50363884070515

-	Q34V, E42P, D24S, E3R y E66L, ΔΔG = -4.6411832294867565

-	Q34V, E42P y D24S, ΔΔG = -1.4394946452724184

-	Q23E, ΔΔG = -0.3439148372373495

-	Q23G, ΔΔG = -0.11714910046833893

-	E3R y A46E, ΔΔG = -1.7318383586820403

Mutaciones recogidas en Wunderlich *et al*. (2005) y sus combinaciones:

-	M1R, E3K, K65I y E66L, ΔΔG = -6.703335106589975

-	M1R, E3K, Q34V, E42P, K65I y E66L, ΔΔG = -9.358061212559392, Δfa_rep = 5.650321498359283

-	M1R, E3R, Q34V, E42P, K65I y E66L, ΔΔG = -10.774280782797518, Δfa_rep = 4.2882533220767485

-	M1R, E3K, D24S, Q34V, E42P, K65I y E66L, ΔΔG = -9.910899215616752

-	M1R, E3R, D24S, Q34V, E42P, K65I y E66L, ΔΔG = -12.050533485250412

-	M1R, E3R, D24S, E42P, K65I y E66L, ΔΔG = -12.009456153025326, Δfa_rep = 2.677577990700293

-	M1R, E3R, D24S, E42P, A46G, K65I y E66L, ΔΔG = -8.185444273934593

-	M1R, E3R, D24S, E42P, A46K, K65I y E66L, ΔΔG = -11.13520692705282

-	M1R, E3K, G14L, D24S, 34V, E42P, K65I y E66L, ΔΔG = -7.293020681513525

-	M1R, E3K, G14L, Q34V, E42P, K65I y E66L, ΔΔG = -7.641389160229323

Mutaciones recogidas en la Tabla 4 de Wunderlich *et al.* (2005) y sus combinaciones:

- M1R, E3R, D24S, E42P, A46K, S48R, K65I y E66L, ΔΔG = -14.781372451660104, Δfa_rep = 3.659892643876198

- M1R, E3V, D24S, E42P, A46L, S48R, T64R, K65I, E66V, A67G, ΔΔG = -16.931475685309607

- M1R, E3R, D24S, E42P, A46L, S48R, T64R, K65I, E66L, A67G, ΔΔG = -13, Δfa_rep = 1,5

- M1R, E3V, D24S, E42P, A46L, S48R, T64R, K65I, E66L, A67G, ΔΔG = -18

- L2R, E3V, A46L, T64R, E66V, A67G, ΔΔG = -12

- L2R, E3V, D24S, A46L, T64R, E66V, A67G, ΔΔG = -14

- L2R, E3V, D24S, A46L, S48R, T64R, E66V, A67G, ΔΔG = -19, Δfa_rep = 1,16

- L2R, E3V, D24S, A46L, S48R, T64R, K65I, E66V, A67G, ΔΔG = -22,9, Δfa_rep = 0,72

- L2R, E3V, D24S, A46L, S48R, T64R, K65I, E66L, A67G, ΔΔG = -18,7 (d)

- L2R, E3R, D24S, A46L, S48R, T64R, K65I, E66L, A67G, ΔΔG = -19

- L2R, E3R, D24S, A46L, S48R, T64R, K65I, E66V, A67G, ΔΔG = -19

- M1R, L2R, E3V, D24S, A46L, S48R, T64R, K65I, E66V, A67G, ΔΔG = -19, Δfa_rep = 1 (c)

- L2R, E3V, D24S, E42P, A46L, S48R, T64R, K65I, E66V, A67G, ΔΔG = -16, Δfa_rep = 2,8 (a)

-	M1R, E3V, D24S, A46L, S48R, T64R, K65I, E66L, A67G, ΔΔG = -17 (b)

Otras mutaciones descartadas:

- G16L, ΔΔG = 7.865124463038427

- 4E y 8E, ΔΔG = 9.32027714574491

- 12P y 15Y, ΔΔG = 9.010892248007508

- 12P, ΔΔG = 8.977791074589163

- 3I, ΔΔG = -0.46165591405804207

- 14L, ΔΔG = -2.659595508509625

- 23I, ΔΔG = -0.7151744018464683

- 30F, ΔΔG = -0.33249119966913554


Diseños seleccionados:

- Diseño 1: M1R, E3R, D24S, E42P, K65I, E66L

- Diseño 2: M1R, E3V, D24S, E42P, A46L, S48R, T64R, K65I, E66L, A67G

- Diseño 3: L2R, E3V, D24S, A46L, S48R, T64R, K65I, E66V, A67G


**Diseño 1: la primera iteración**

Mediante la consulta de bibliografía relacionada con la modificación de proteínas para la mejora de su estabilidad térmica y sobre Bs-CspB en concreto, se han concretado varias mutaciones que pueden encontrarse en la proteína de la variante termófila de Bacillus subtilis como la mutación a serina en la posición 24 (Mueller *et al*., 2000) y otras mutaciones predichas mediante programas informáticos como la mutación a isoleucina en la posición 65 (Wunderlich *et al*., 2005). La aportación de cada cambio a la estabilidad global de la proteína se explica a continuación según la información disponible en la literatura y las conclusiones obtenidas en este experimento.

La mutación generada en la posición 1 (metionina--> arginina) proporciona una estructura nueva con interacciones estabilizadoras adicionales, gracias a la introducción de la carga positiva por parte de la arginina. Además, se caracteriza porque la estructura resultante presenta un mejor empaquetamiento de esas cadenas laterales y nuevas interacciones de carga (Wunderlich *et al*., 2005).

La mutación en la posición 3 (glutamato (negativa)-->arginina (positiva)) es muy favorable por la contribución electrostática de la arginina, que puede participar en fuertes interacciones electrostáticas, como los puentes salinos, los cuales favorecen un mejor empaquetamiento de determinadas zonas de la proteína y aportan una mayor rigidez a altas temperaturas (Perl *et al*., 2000).

La mutación de la posición 24 (aspartato--> serina) carece de demostración experimental directa, pero sí se puede determinar que la sustitución por pequeños residuos polares, como es la serina, consigue estabilizar las interacciones locales sin alterar la geometría estructural (Schindelin *et al*., 1993).

La mutación en la posición 42 (glutamato--> prolina) se caracteriza porque la introducción de un residuo de prolina en un bucle flexible de la estructura puede estabilizar la proteína al aumentar la estabilidad termodinámica gracias a la restricción de entropía del desorden de la cadena desplegada (Watanabe *et al*., 1998). En la actualidad, esta mutación no ha sido probada experimentalmente en nuestra proteína.

La mutación introducida en la posición 65 (lisina (positiva)--> isoleucina (neutra/hidrofóbica)) genera una estructura con un mejor empaquetamiento. Se produce la sustitución o "enterramiento" de la carga positiva de la leucina por una cadena lateral alifática que se empaqueta mejor con el núcleo hidrofóbico de la proteína (Max *et al*., 2007).

La mutación en la posición 66 (glutamato (positiva)--> leucina (neutral/hidrofóbica)), al igual que la mutación en la posición 3, se realiza porque se ha identificado como aminoácido esencial en la estabilidad. La estabilidad de la mutación a leucina radica en la hidrofobicidad de la leucina, lo que favorece interacciones hidrofóbicas. Además, se encuentra en la región C-terminal de la proteína y al aumentar las interacciones en una región terminal de la proteína, se mejora el empaquetamiento de la estructura, lo que disminuye la posibilidad de que se desnaturalice por calor (Perl *et al*., 2000).


In [ ]:
# nodesign resfile
resfile_content = """
NATRO
start
1 A PIKAA R
3 A PIKAA R
24 A PIKAA S
42 A PIKAA P
65 A PIKAA I
66 A PIKAA L
"""

# Se define la ruta en la que se guardará el refile
output_resfile_path = "/content/design1.resfile"

# Se añade contenido al resfile
with open(output_resfile_path, "w") as f:
    f.write(resfile_content)

print(f"Resfile saved to: {output_resfile_path}")

Resfile saved to: /content/design1.resfile


In [ ]:
# new design resfile
resfile = "/content/design1.resfile"

# Se resetea a la conformación incial
pose_design1 = full_pdb_pose.clone()

tf = pyrosetta.rosetta.core.pack.task.TaskFactory()

tf.push_back(pyrosetta.rosetta.core.pack.task.operation.InitializeFromCommandline())
tf.push_back(pyrosetta.rosetta.core.pack.task.operation.IncludeCurrent())


# Se incluye el resfile
tf.push_back(pyrosetta.rosetta.core.pack.task.operation.ReadResfile(resfile))

packer_task = tf.create_task_and_apply_taskoperations(pose_design1)
print(packer_task)


mm = pyrosetta.rosetta.core.kinematics.MoveMap()
mm.set_bb(True)
mm.set_chi(True)
mm.set_jump(True)

scorefxn = pyrosetta.create_score_function("ref2015_cart.wts")
rel_design = pyrosetta.rosetta.protocols.relax.FastRelax(scorefxn_in=scorefxn, standard_repeats=1, script_file="MonomerDesign2019")
rel_design.cartesian(True)
rel_design.set_task_factory(tf)
rel_design.set_movemap(mm)
rel_design.minimize_bond_angles(True)
rel_design.minimize_bond_lengths(True)

%time rel_design.apply(pose_design1)

core.pack.task: Packer task: initialize from command line()
#Packer_Task

Threads to request: ALL AVAILABLE

resid	pack?	design?	allowed_aas
1	TRUE	TRUE	ARG:NtermProteinFull
2	FALSE	FALSE	
3	TRUE	TRUE	ARG
4	FALSE	FALSE	
5	FALSE	FALSE	
6	FALSE	FALSE	
7	FALSE	FALSE	
8	FALSE	FALSE	
9	FALSE	FALSE	
10	FALSE	FALSE	
11	FALSE	FALSE	
12	FALSE	FALSE	
13	FALSE	FALSE	
14	FALSE	FALSE	
15	FALSE	FALSE	
16	FALSE	FALSE	
17	FALSE	FALSE	
18	FALSE	FALSE	
19	FALSE	FALSE	
20	FALSE	FALSE	
21	FALSE	FALSE	
22	FALSE	FALSE	
23	FALSE	FALSE	
24	TRUE	TRUE	SER
25	FALSE	FALSE	
26	FALSE	FALSE	
27	FALSE	FALSE	
28	FALSE	FALSE	
29	FALSE	FALSE	
30	FALSE	FALSE	
31	FALSE	FALSE	
32	FALSE	FALSE	
33	FALSE	FALSE	
34	FALSE	FALSE	
35	FALSE	FALSE	
36	FALSE	FALSE	
37	FALSE	FALSE	
38	FALSE	FALSE	
39	FALSE	FALSE	
40	FALSE	FALSE	
41	FALSE	FALSE	
42	TRUE	TRUE	PRO
43	FALSE	FALSE	
44	FALSE	FALSE	
45	FALSE	FALSE	
46	FALSE	FALSE	
47	FALSE	FALSE	
48	FALSE	FALSE	
49	FALSE	FALSE	
50	FALSE	FALSE	
51	FALSE	FALSE	
52	FALSE	FALSE	
53	FALSE	FALSE	

Se analiza el diseño siguiendo las mismas pautas que para la estructura original con el objetivo de, posteriormente, realizar la comparación  y se recopila en "metricas_design1":

In [ ]:
print("ANÁLISIS DISEÑO 1:")
print()
metricas_design1 = calcular_metricas_evaluacion(
    nombre_diseno="Diseño 1",
    pose_wt=pose_relax,
    pose_diseno=pose_design1,
    tm_wt=TM_WILD_TYPE
)

# Alineamiento
_, alineamiento_str = obtener_mutaciones_y_alineamiento(pose_relax, pose_design1)
print(alineamiento_str)

# Guarda la info en la lista de resultados
lista_resultados.append(metricas_design1)

# Imprime resumen
print(f"RMSD vs WT: {metricas_design1['RMSD (Å)']:.3f} Å")
print(f"ΔΔG: {metricas_design1['ΔΔG (REU)']:.2f} REU")
print(f"Δfa_rep: {metricas_design1['Δfa_rep']:.2f} REU")
print(f"Tm Estimado: {metricas_design1['Tm Diseño (°C)']:.1f} °C")
print(f"Puentes Salinos: {metricas_design1['Dis_Nº Puentes Salinos']}")

ANÁLISIS DISEÑO 1:

Alineamiento de Secuencia:
  WT:     1    MLEGKVKWFNSEKGFGFIEVEGQDDVFVHFSAIQGEGFKTLEEGQAVSFEIVEGNRGPQA
                | |||||||||||||||||||| ||||||||||||||||| ||||||||||||||||||
  Design: 1    RLRGKVKWFNSEKGFGFIEVEGQSDVFVHFSAIQGEGFKTLPEGQAVSFEIVEGNRGPQA

  WT:     61   ANVTKEA
               ||||  |
  Design: 61   ANVTILA

Mutaciones aplicadas (6): M1R, E3R, D24S, E42P, K65I, E66L
RMSD vs WT: 0.853 Å
ΔΔG: -12.01 REU
Δfa_rep: 2.68 REU
Tm Estimado: 134.3 °C
Puentes Salinos: 1


Tras las modificaciones, la secuencia de la estructura diseñada sería la siguiente:

In [ ]:
print(pose_design1.sequence())

RLRGKVKWFNSEKGFGFIEVEGQSDVFVHFSAIQGEGFKTLPEGQAVSFEIVEGNRGPQAANVTILA


Se comprueba que las mutaciones no eliminan el puente salino de la estructura original y se observa si se generan otros nuevos. Además, se analizan los puentes de hidrógeno de la nueva estructura:

In [ ]:
puentes_salinos_design1 = puentes_salinos(pose_design1, SB_CUTOFF_DISTANCE)
design1_sb_count = puentes_salinos_design1['Nº de puentes salinos']
print(f"Puentes salinos en design1: {design1_sb_count} puentes salinos encontrados.")
print("Los puentes salinos se dan entre los aminoácidos:")
for sb in puentes_salinos_design1["Lista de puentes salinos"]:
    print(sb)

Puentes salinos en design1: 1 puentes salinos encontrados.
Los puentes salinos se dan entre los aminoácidos:
  7 LYS <->  25 ASP  (3.65 Å)


In [ ]:
hbond_info1 = puentes_hidrogeno(pose_design1)

print("Análisis de Puentes de Hidrógeno:")
print()
print(f"  Total: {hbond_info1['Nº H-Bonds (Total)']}")
print(f"  Backbone-Backbone: {hbond_info1['Nº H-Bonds (BB)']}")
print(f"  Sidechain/Mixtos: {hbond_info1['Nº H-Bonds (SC)']}")

Análisis de Puentes de Hidrógeno:

  Total: 48
  Backbone-Backbone: 37
  Sidechain/Mixtos: 11


In [ ]:
from pyrosetta.rosetta.core.scoring import CA_rmsd

# Dentro del bucle, después de tener 'pose_relax' y 'pose_design1'
rmsd_valor = CA_rmsd(pose_relax, pose_design1)

# Se busca un valor bajo, idealmente < 1.0 Å, aceptable hasta 2.0 Å

**Diseño 2: la evolución del diseño**

Tras la consulta de bibliografía adicional y las pruebas realizadas sobre el diseño anterior y distintas combinaciones del mismo, se han encontrado nuevos cambios dirigidos hacia el incremento de la estabilidad térmica. En este diseño, se añaden las mutaciones A46L, S48R, T64R y A67G y se modifica la mutación introducida anteriormente en la posición 3 al aminoácido valina.

La mutación a arginina en la posición 3 explorada anteriormente solo resulta favorable según el contexto en el que se introduce. En diseños que también usan las mutaciones a arginina en la posición 48 y a leucina en la posición 66, se ha concluido mediante modelos realizados por ordenador que esta sustitución no resulta tan estabilizadora (Wunderlich *et al*, 2005). Por otro lado, la mutación generada en la posición 3 (glutamato-->valina) carece de respaldo experimental directo, pero sí se ha podido determinar que la posición 3 es esencial en la estabilidad térmica y queda en la superficie de la proteína. El cambio mejora la química local de ese sitio, ya sea aumentando la hidrofobicidad en la superficie o permitiendo un empaquetamiento más eficiente en el núcleo, lo que resulta en una disminución de la energía libre de Gibbs (Diaz *et al*., 2024).

La mutación en la posición 46 (alanina-->leucina) se caracteriza porque produce una mejora en las interacciones hidrofóbicas. Se sugiere que la leucina en la posición 46 se relaciona mediante interacción hidrofóbica con Val66 en la variante estabilizada (Martin *et al*., 2002).

La mutación en la posición 48 (serina -->arginina) reduce la carga neta, lo que favorece la estabilización. Esta mutación genera nuevas interacciones y atenúa interacciones desestabilizadoras ya existentes. Este residuo positivamente cargado ayuda a compensar el par de residuos cargados negativamente Glu3 y Glu66 (Max et al., 2007). También contribuye al mecanismo general de estabilización a través de interacciones de carga, enlaces de hidrógeno adicionales y un mejor empaquetamiento de la cadena lateral (Max *et al*., 2007).

La mutación en la posición 64 (treonina-->arginina) aporta estabilidad centrándose principalmente en las interacciones de carga al introducir un residuo de carga positiva en la superficie de la estructura. Además, disminuye la carga neta de la proteína. También cabe destacar que la Arg64 es capaz de hacer desaparecer la repulsión entre Glu3 y Glu66 en la estructura original (Martin et al., 2002) (Wunderlich *et al*., 2005).

La mutación en la posición 67 (alanina-->glicina) se caracteriza porque es una mutación en la última posición, que puede tener efectos en el plegamiento y estabilidad, aunque no tenga evidencia experimental directa. Se ha probado con diferentes aminoácidos en esta posición, todos ellos pequeños y sin carga. De las estructuras establecidas, aquellas en las que el residuo 67 estaba ocupado por una glicina, presentaban mayor estabilidad (Martin *et al*., 2002).


In [ ]:
# nodesign resfile
resfile_content = """
NATRO
start
1 A PIKAA R
3 A PIKAA V
24 A PIKAA S
42 A PIKAA P
46 A PIKAA L
48 A PIKAA R
64 A PIKAA R
65 A PIKAA I
66 A PIKAA L
67 A PIKAA G
"""

# Define la ruta en la que se guardará el refile
output_resfile_path = "/content/design2.resfile"

# Contenido al resfile
with open(output_resfile_path, "w") as f:
    f.write(resfile_content)

print(f"Resfile saved to: {output_resfile_path}")

Resfile saved to: /content/design2.resfile


In [ ]:
# new design resfile
resfile = "/content/design2.resfile"

# Resetea a la conformación incial
pose_design2 = full_pdb_pose.clone()


tf = pyrosetta.rosetta.core.pack.task.TaskFactory()

tf.push_back(pyrosetta.rosetta.core.pack.task.operation.InitializeFromCommandline())
tf.push_back(pyrosetta.rosetta.core.pack.task.operation.IncludeCurrent())


# Se incluye el resfile
tf.push_back(pyrosetta.rosetta.core.pack.task.operation.ReadResfile(resfile))

packer_task = tf.create_task_and_apply_taskoperations(pose_design2)
print(packer_task)


mm = pyrosetta.rosetta.core.kinematics.MoveMap()
mm.set_bb(True)
mm.set_chi(True)
mm.set_jump(True)

scorefxn = pyrosetta.create_score_function("ref2015_cart.wts")
rel_design = pyrosetta.rosetta.protocols.relax.FastRelax(scorefxn_in=scorefxn, standard_repeats=1, script_file="MonomerDesign2019")
rel_design.cartesian(True)
rel_design.set_task_factory(tf)
rel_design.set_movemap(mm)
rel_design.minimize_bond_angles(True)
rel_design.minimize_bond_lengths(True)

%time rel_design.apply(pose_design2)

core.pack.task: Packer task: initialize from command line()
#Packer_Task

Threads to request: ALL AVAILABLE

resid	pack?	design?	allowed_aas
1	TRUE	TRUE	ARG:NtermProteinFull
2	FALSE	FALSE	
3	TRUE	TRUE	VAL
4	FALSE	FALSE	
5	FALSE	FALSE	
6	FALSE	FALSE	
7	FALSE	FALSE	
8	FALSE	FALSE	
9	FALSE	FALSE	
10	FALSE	FALSE	
11	FALSE	FALSE	
12	FALSE	FALSE	
13	FALSE	FALSE	
14	FALSE	FALSE	
15	FALSE	FALSE	
16	FALSE	FALSE	
17	FALSE	FALSE	
18	FALSE	FALSE	
19	FALSE	FALSE	
20	FALSE	FALSE	
21	FALSE	FALSE	
22	FALSE	FALSE	
23	FALSE	FALSE	
24	TRUE	TRUE	SER
25	FALSE	FALSE	
26	FALSE	FALSE	
27	FALSE	FALSE	
28	FALSE	FALSE	
29	FALSE	FALSE	
30	FALSE	FALSE	
31	FALSE	FALSE	
32	FALSE	FALSE	
33	FALSE	FALSE	
34	FALSE	FALSE	
35	FALSE	FALSE	
36	FALSE	FALSE	
37	FALSE	FALSE	
38	FALSE	FALSE	
39	FALSE	FALSE	
40	FALSE	FALSE	
41	FALSE	FALSE	
42	TRUE	TRUE	PRO
43	FALSE	FALSE	
44	FALSE	FALSE	
45	FALSE	FALSE	
46	TRUE	TRUE	LEU
47	FALSE	FALSE	
48	TRUE	TRUE	ARG
49	FALSE	FALSE	
50	FALSE	FALSE	
51	FALSE	FALSE	
52	FALSE	FALSE	
53	FALSE	FALS

Se analiza este diseño y se recopila toda la información en "metricas_design2":

In [ ]:
print("ANÁLISIS DISEÑO 2:")

metricas_design2 = calcular_metricas_evaluacion(
    nombre_diseno="Diseño 2",
    pose_wt=pose_relax,
    pose_diseno=pose_design2,
    tm_wt=TM_WILD_TYPE
)

# Alineamiento
_, alineamiento_str_2 = obtener_mutaciones_y_alineamiento(pose_relax, pose_design2)
print(alineamiento_str_2)

# Se añade la info a la lista de resultados
lista_resultados.append(metricas_design2)

# Imprime un resumen
print(f"RMSD vs WT: {metricas_design2['RMSD (Å)']:.3f} Å")
print(f"ΔΔG: {metricas_design2['ΔΔG (REU)']:.2f} REU")
print(f"Δfa_rep: {metricas_design2['Δfa_rep']:.2f} REU")
print(f"Tm Estimado: {metricas_design2['Tm Diseño (°C)']:.1f} °C")
print(f"Puentes Salinos: {metricas_design2['Dis_Nº Puentes Salinos']}")


ANÁLISIS DISEÑO 2:
Alineamiento de Secuencia:
  WT:     1    MLEGKVKWFNSEKGFGFIEVEGQDDVFVHFSAIQGEGFKTLEEGQAVSFEIVEGNRGPQA
                | |||||||||||||||||||| ||||||||||||||||| ||| | ||||||||||||
  Design: 1    RLVGKVKWFNSEKGFGFIEVEGQSDVFVHFSAIQGEGFKTLPEGQLVRFEIVEGNRGPQA

  WT:     61   ANVTKEA
               |||    
  Design: 61   ANVRILG

Mutaciones aplicadas (10): M1R, E3V, D24S, E42P, A46L, S48R, T64R, K65I, E66L, A67G
RMSD vs WT: 0.690 Å
ΔΔG: -18.39 REU
Δfa_rep: 1.27 REU
Tm Estimado: 179.5 °C
Puentes Salinos: 1


Se calculan los puentes salinos y los puentes de hidrógeno de la estructura mutada.

In [ ]:
puentes_salinos_design2 = puentes_salinos(pose_design2, SB_CUTOFF_DISTANCE)
design2_sb_count = puentes_salinos_design2['Nº de puentes salinos']
print(f"Puentes salinos en design2: {design2_sb_count} puentes salinos encontrados.")
print("Los puentes salinos se dan entre los aminoácidos:")
for sb in puentes_salinos_design2["Lista de puentes salinos"]:
    print(sb)

Puentes salinos en design2: 1 puentes salinos encontrados.
Los puentes salinos se dan entre los aminoácidos:
  7 LYS <->  25 ASP  (3.64 Å)


In [ ]:
hbond_info2 = puentes_hidrogeno(pose_design2)

print("Análisis de Puentes de Hidrógeno:")
print()
print(f"  Total: {hbond_info2['Nº H-Bonds (Total)']}")
print(f"  Backbone-Backbone: {hbond_info2['Nº H-Bonds (BB)']}")
print(f"  Sidechain/Mixtos: {hbond_info2['Nº H-Bonds (SC)']}")

Análisis de Puentes de Hidrógeno:

  Total: 46
  Backbone-Backbone: 36
  Sidechain/Mixtos: 10


**Diseño final: la optimización**

Tras obtener buenos resultados en los diseños anteriores, se buscó la optimización de la estructura mediante variaciones en las mutaciones seleccionadas. En este diseño, se ha modificado la mutación a arginina de la posición 1 a la posición 2 y se ha sustituido la mutación a leucina en la posición 66 por una mutación a valina (Wunderlich *et al*., 2005). Se ha comprobado experimentalmente que la mutación a prolina en la posición 42 no resulta favorable en este contexto, por lo que se ha eliminado de este diseño (diseño señalado como a en la lista).

La mutación generada en la posición 2 (leucina-->arginina) no se ha podido determinar experimentalmente como es el caso de 1R y 3R. Sin embargo, se ha conseguido demostrar que la introducción de aminoácidos cargados positivamente, como es la arginina, en la región N-terminal (aminoácidos 1, 2 y 3), mejora la estabilidad global de la proteína. Esta mayor estabilidad se consigue mediante la optimización de interacciones electrostáticas (Wunderlich *et al*., 2005). Mediante pruebas por ordenador realizadas para este trabajo, se ha concluido que el diseño era más estable conteniendo arginina en la posición 2 comparado con el mismo aminoácido en la posición 1 o en ambas (diseños señalados como b y c, respectivamente).

La mutación introducida en la posición 66 (glutamato-->valina) no se ha determinado experimentalmente de forma directa, más bien se ha tenido en cuenta su parecido a la leucina. El intercambio de un aminoácido polar o negativo por un aminoácido hidrofóbico mejora la estabilidad al eliminar fuerzas de repulsión y optimizar el empaquetamiento hidrofóbico (Wunderlich *et al*., 2005). Además, se ha podido determinar experimentalmente que la sustitución en las posiciones 65 y 66 por un residuo apolar genera un mejor empaquetamiento (Max *et al*., 2007). Mediante pruebas por ordenador realizadas para este trabajo, se ha concluido que el diseño disponía de menos estabilidad si se continuaba con el aminoácido leucina en esta posición (diseño señalado como d).


In [ ]:
# nodesign resfile
resfile_content = """
NATRO
start
2 A PIKAA R
3 A PIKAA V
24 A PIKAA S
46 A PIKAA L
48 A PIKAA R
64 A PIKAA R
65 A PIKAA I
66 A PIKAA V
67 A PIKAA G
"""

# Define la ruta donde se guardará el resfile
output_resfile_path = "/content/design3.resfile"

# Contenido al resfile
with open(output_resfile_path, "w") as f:
    f.write(resfile_content)

print(f"Resfile saved to: {output_resfile_path}")

Resfile saved to: /content/design3.resfile


In [ ]:
# new design resfile
resfile = "/content/design3.resfile"

# Resetea a la conformación incial
pose_design3 = full_pdb_pose.clone()


tf = pyrosetta.rosetta.core.pack.task.TaskFactory()

tf.push_back(pyrosetta.rosetta.core.pack.task.operation.InitializeFromCommandline())
tf.push_back(pyrosetta.rosetta.core.pack.task.operation.IncludeCurrent())


# Se incluye el resfile
tf.push_back(pyrosetta.rosetta.core.pack.task.operation.ReadResfile(resfile))

packer_task = tf.create_task_and_apply_taskoperations(pose_design3)
print(packer_task)


mm = pyrosetta.rosetta.core.kinematics.MoveMap()
mm.set_bb(True)
mm.set_chi(True)
mm.set_jump(True)

scorefxn = pyrosetta.create_score_function("ref2015_cart.wts")
rel_design = pyrosetta.rosetta.protocols.relax.FastRelax(scorefxn_in=scorefxn, standard_repeats=1, script_file="MonomerDesign2019")
rel_design.cartesian(True)
rel_design.set_task_factory(tf)
rel_design.set_movemap(mm)
rel_design.minimize_bond_angles(True)
rel_design.minimize_bond_lengths(True)

%time rel_design.apply(pose_design3)

core.pack.task: Packer task: initialize from command line()
#Packer_Task

Threads to request: ALL AVAILABLE

resid	pack?	design?	allowed_aas
1	FALSE	FALSE	
2	TRUE	TRUE	ARG
3	TRUE	TRUE	VAL
4	FALSE	FALSE	
5	FALSE	FALSE	
6	FALSE	FALSE	
7	FALSE	FALSE	
8	FALSE	FALSE	
9	FALSE	FALSE	
10	FALSE	FALSE	
11	FALSE	FALSE	
12	FALSE	FALSE	
13	FALSE	FALSE	
14	FALSE	FALSE	
15	FALSE	FALSE	
16	FALSE	FALSE	
17	FALSE	FALSE	
18	FALSE	FALSE	
19	FALSE	FALSE	
20	FALSE	FALSE	
21	FALSE	FALSE	
22	FALSE	FALSE	
23	FALSE	FALSE	
24	TRUE	TRUE	SER
25	FALSE	FALSE	
26	FALSE	FALSE	
27	FALSE	FALSE	
28	FALSE	FALSE	
29	FALSE	FALSE	
30	FALSE	FALSE	
31	FALSE	FALSE	
32	FALSE	FALSE	
33	FALSE	FALSE	
34	FALSE	FALSE	
35	FALSE	FALSE	
36	FALSE	FALSE	
37	FALSE	FALSE	
38	FALSE	FALSE	
39	FALSE	FALSE	
40	FALSE	FALSE	
41	FALSE	FALSE	
42	FALSE	FALSE	
43	FALSE	FALSE	
44	FALSE	FALSE	
45	FALSE	FALSE	
46	TRUE	TRUE	LEU
47	FALSE	FALSE	
48	TRUE	TRUE	ARG
49	FALSE	FALSE	
50	FALSE	FALSE	
51	FALSE	FALSE	
52	FALSE	FALSE	
53	FALSE	FALSE	
54	FALSE	FALSE	

Se recopila la información extraida del diseño final en "metricas_design3":


In [ ]:
print("ANÁLISIS DISEÑO FINAL:")
print()

# Cálculo de métricas
metricas_design3 = calcular_metricas_evaluacion(
    nombre_diseno="Diseño Final",
    pose_wt=pose_relax,
    pose_diseno=pose_design3,
    tm_wt=TM_WILD_TYPE
)

# Alineamiento
_, alineamiento_str_3 = obtener_mutaciones_y_alineamiento(pose_relax, pose_design3)
print(alineamiento_str_3)

# ASe añade la info a la lista de resultados
lista_resultados.append(metricas_design3)

# Imprime un resumen
print(f"RMSD vs WT: {metricas_design3['RMSD (Å)']:.3f} Å")
print(f"ΔΔG: {metricas_design3['ΔΔG (REU)']:.2f} REU")
print(f"Δfa_rep: {metricas_design3['Δfa_rep']:.2f} REU")
print(f"Tm Estimado: {metricas_design3['Tm Diseño (°C)']:.1f} °C")
print(f"Puentes Salinos: {metricas_design3['Dis_Nº Puentes Salinos']}")


ANÁLISIS DISEÑO FINAL:

Alineamiento de Secuencia:
  WT:     1    MLEGKVKWFNSEKGFGFIEVEGQDDVFVHFSAIQGEGFKTLEEGQAVSFEIVEGNRGPQA
               |  |||||||||||||||||||| ||||||||||||||||||||| | ||||||||||||
  Design: 1    MRVGKVKWFNSEKGFGFIEVEGQSDVFVHFSAIQGEGFKTLEEGQLVRFEIVEGNRGPQA

  WT:     61   ANVTKEA
               |||    
  Design: 61   ANVRIVG

Mutaciones aplicadas (9): L2R, E3V, D24S, A46L, S48R, T64R, K65I, E66V, A67G
RMSD vs WT: 1.027 Å
ΔΔG: -22.96 REU
Δfa_rep: 0.72 REU
Tm Estimado: 212.0 °C
Puentes Salinos: 1


Se calulan los puentes salinos y los puentes de hidrógeno.

In [ ]:
puentes_salinos_design3 = puentes_salinos(pose_design3, SB_CUTOFF_DISTANCE)
design3_sb_count = puentes_salinos_design3['Nº de puentes salinos']
print(f"Puentes salinos en design3: {design3_sb_count} puentes salinos encontrados.")
print("Los puentes salinos se dan entre los aminoácidos:")
for sb in puentes_salinos_design3["Lista de puentes salinos"]:
    print(sb)

Puentes salinos en design3: 1 puentes salinos encontrados.
Los puentes salinos se dan entre los aminoácidos:
  7 LYS <->  25 ASP  (3.63 Å)


In [ ]:
hbond_info3 = puentes_hidrogeno(pose_design3)

print("Análisis de Puentes de Hidrógeno:")
print()
print(f"  Total: {hbond_info3['Nº H-Bonds (Total)']}")
print(f"  Backbone-Backbone: {hbond_info3['Nº H-Bonds (BB)']}")
print(f"  Sidechain/Mixtos: {hbond_info3['Nº H-Bonds (SC)']}")

Análisis de Puentes de Hidrógeno:

  Total: 46
  Backbone-Backbone: 37
  Sidechain/Mixtos: 9


# 3. RESULTADOS
**Métricas de evaluación y análisis**


Con la información recogida de los distintos diseños y de la proteína original, se puede elaborar una tabla comparativa:

In [ ]:
print("TABLA COMPARATIVA DE DISEÑOS")

# Convertimos la lista de resultados que hemos ido cogiendo en una tabla de Pandas
df_resultados = pd.DataFrame(lista_resultados)

# Definimos las columnas que queremos ver y su orden
columnas_display = [
    "Diseño",
    "Score Estabilidad",
    "ΔTm Estimado (°C)",
    "Tm Diseño (°C)",
    "ΔΔG (REU)",
    "RMSD (Å)",
    "Nº Mutaciones",
    "Δfa_rep",
    "Bonus SB",
    "Dis_Nº Puentes Salinos",
    "Dis_Nº H-Bonds (Total)"
]
# Mostramos la tabla ordenada por el mejor "Score Estabilidad"
display(
    df_resultados[columnas_display].sort_values("Score Estabilidad").reset_index(drop=True)
)

TABLA COMPARATIVA DE DISEÑOS


,Diseño,Score Estabilidad,ΔTm Estimado (°C),Tm Diseño (°C),ΔΔG (REU),RMSD (Å),Nº Mutaciones,Δfa_rep,Bonus SB,Dis_Nº Puentes Salinos,Dis_Nº H-Bonds (Total)
0,Diseño Final,-18.369070,163.025492,212.025492,-22.961337,1.027108,9,0.720641,No,1,46
1,Diseño 2,-14.708751,130.540166,179.540166,-18.385939,0.690414,10,1.268206,No,1,46
2,Diseño 1,-10.808511,85.267139,134.267139,-12.009456,0.853392,6,2.677578,No,1,48
3,Wild-Type,0.000000,-0.000000,49.000000,0.000000,0.000000,0,0.000000,No,1,44


Se puede ver claramente en la tabla que todos los diseños estudiados son exitosos en cuanto a los requisitos de ΔΔG (< -2 REU) y de Δfa_rep (< 20 REU). Por último, se muestran las estructuras mutadas de forma visual para comprobar que mantienen la estructura de barril beta característica de esta proteína.

In [ ]:
import py3Dmol

# List of poses and their names to display
poses_to_display = [
    (pose_relax, "Wild-Type"),
    (pose_design1, "Diseño 1"),
    (pose_design2, "Diseño 2"),
    (pose_design3, "Diseño Final")
]

# Create a view for each protein structure
for pose, name in poses_to_display:
    # Save the pose to a temporary PDB file
    temp_pdb_filepath = f"{name.replace(' ', '_')}_temp.pdb"
    pose.dump_pdb(temp_pdb_filepath)

    # Read the PDB data from the temporary file
    with open(temp_pdb_filepath) as f:
        pdb_data = f.read()

    # Create and display the 3Dmol view
    view = py3Dmol.view(width=400, height=400)
    view.addModel(pdb_data, "pdb")
    view.setStyle({"cartoon": {"color": "spectrum"}})  # Use spectrum color for cartoon
    view.zoomTo()
    view.addLabel(name, {'position': {'x': 0, 'y': 0, 'z': 0}, 'fontColor':'black', 'backgroundColor': 'white'}) # Add a label for the design name
    view.show()

    # Clean up the temporary PDB file
    os.remove(temp_pdb_filepath)

3Dmol.js failed to load for some reason. Please check your browser console for error messages.

3Dmol.js failed to load for some reason. Please check your browser console for error messages.

3Dmol.js failed to load for some reason. Please check your browser console for error messages.

3Dmol.js failed to load for some reason. Please check your browser console for error messages.

Se muestra a continuación una función que permite observar claramente las diferencias estructurales entre la proteína original y el diseño final. Esta función alinea las dos poses, una móvil (diseño final) sobre una de referencia (original) y las muestra superpuestas con colores diferentes:

In [ ]:
import py3Dmol
import tempfile
from pyrosetta import Pose
from pyrosetta.rosetta.core.scoring import calpha_superimpose_pose

def display_superposition(pose_ref: Pose, pose_mobile: Pose, title_ref: str, title_mobile: str):
    """
    - pose_ref (Original) se mostrará en azul.
    - pose_mobile (Diseño) se mostrará en verde.
    """
    print(f"Original vs. Diseño final: {title_ref} (Azul) vs. {title_mobile} (Verde)")

    # Clona la pose móvil para no modificar la original
    mobile_aligned = Pose()
    mobile_aligned.assign(pose_mobile)

    # Alinea la pose 'móvil' sobre la pose 'referencia'
    # Esto calcula el RMSD y mueve la pose 'mobile_aligned'
    calpha_superimpose_pose(mobile_aligned, pose_ref)

    # Guarda las poses en archivos temporales para py3Dmol
    with tempfile.NamedTemporaryFile(suffix=".pdb") as tmp_ref, \
         tempfile.NamedTemporaryFile(suffix=".pdb") as tmp_mobile:

        pose_ref.dump_pdb(tmp_ref.name)
        pdb_str_ref = open(tmp_ref.name).read()

        mobile_aligned.dump_pdb(tmp_mobile.name)
        pdb_str_mobile = open(tmp_mobile.name).read()

    # Configura la vista de py3Dmol
    view = py3Dmol.view(width=800, height=500)

    # Añade el Modelo 0: Referencia (WT) en azul
    # Usamos opacidad para poder ver el otro modelo a través de él
    view.addModel(pdb_str_ref, "pdb")
    view.setStyle({'model': 0}, {"cartoon": {"color": "lightblue", "opacity": 0.8}})
    view.addLabel(title_ref + " (Azul)", {'model': 0, 'position': {'x': 0, 'y': 20, 'z': 0}, 'fontColor':'lightblue', 'backgroundColor': 'white'})

    # Añade el Modelo 1: Diseño (Mutante) en verde
    view.addModel(pdb_str_mobile, "pdb")
    view.setStyle({'model': 1}, {"cartoon": {"color": "green"}})
    view.addLabel(title_mobile + " (Verde)", {'model': 1, 'position': {'x': 0, 'y': -20, 'z': 0}, 'fontColor':'green', 'backgroundColor': 'white'})

    view.zoomTo()
    view.show()

In [ ]:

display_superposition(pose_relax, pose_design3, "Original", "Diseño final")

Original vs. Diseño final: Original (Azul) vs. Diseño final (verde)


3Dmol.js failed to load for some reason. Please check your browser console for error messages.

# 4. DISCUSIÓN

El conjunto de diseños analizados muestra una mejora progresiva en los parámetros de estabilidad en comparación con la proteína nativa (Wild-Type), lo que demuestra que la introducción racional de mutaciones puede modificar sustancialmente el comportamiento termoestable de Bs-CspB. Los tres diseños expuestos en la tabla de resultados, presentan una reducción en el score de estabilidad y en ΔΔG, así como un aumento en la ΔTm estimada. Estos resultados son indicadores de una mayor estabilidad estructural en comparación con la proteína nativa.

El diseño final destaca como el más favorable atendiendo a los valores de estabilidad. Presenta un score de estabilidad muy inferior al de la proteína nativa, relativizada a un valor de 0. La variación de temperatura estimada es de 163º C, un aumento de 114º C con respecto a la proteína Wild-Type. La variación de energía libre de Gibbs disminuye en el diseño optimizado, lo que indica una disminución de la energía libre del estado plegado. La energía repulsiva aumenta en menos de 1 REU, lo que nos informa sobre las pocas colisiones atómicas, un empaque proteico razonable y que no aparecen tensiones estructurales fuertes. Aunque un elevado número de mutaciones podría conllevar riesgos de perturbaciones estructurales, el RMSD relativamente bajo indica que la arquitectura del barril beta queda conservada. Por tanto, las mutaciones se distribuyeron en posiciones que no comprometen el plegamiento general.

El número de puentes salinos permanece constante en todos los modelos, por lo que las mejoras observadas deben atribuirse al empaquetamiento hidrofóbico, la optimización de las interacciones de Van del Waals y la redistribución de interacciones polares. Además, el aumento del número de enlaces de hidrógeno en los diseños refuerza la estructura, y por tanto, la termoestabilidad.

# 5. CONCLUSIÓN

En conclusión, los resultados computacionales indican que la estabilidad térmica de la proteína de choque térmico frío de la especie *Bacillus subtilis* puede ser incrementada de manera sustancial mediante un diseño racional adecuado. El diseño final optimizado combina una elevada termoestabilidad estimada, un comportamiento energético favorable, conserva la estructura de baril beta y no introduce tensiones estéricas relevantes a nivel atómico. Además, el número de interacciones estabilizadoras como puentes de hidrógeno aumenta, mejorando el empaque estructural, a la vez que el puente salino inicial queda conservado en el diseño final. No obstante, la posterior validación experimental será necesaria para confirmar la efectividad real del diseño.

# 6. BIBLIOGRAFÍA

Brandi, A., et al. (1991). Characterization of cspB, a Bacillus subtilis inducible cold‑shock gene encoding a 67‑amino‑acid protein. Journal of Bacteriology, 173(15), 4889–4892. https://pubmed.ncbi.nlm.nih.gov/1400185/

Diaz, D.J., Gong, C., Ouyang-Zhang, J. et al. Stability Oracle: a structure-based graph-transformer framework for identifying stabilizing mutations. Nat Commun 15, 6170 (2024). https://doi.org/10.1038/s41467-024-49780-

Martin, A., Kather, I., & Schmid, F. X. (2002). Origins of the high stability of an in vitro-selected cold-shock protein. Journal of molecular biology, 318(5), 1341–1349. https://doi.org/10.1016/s0022-2836(02)00243-7

Max, K. E., Wunderlich, M., Roske, Y., Schmid, F. X., & Heinemann, U. (2007). Optimized variants of the cold shock protein from in vitro selection: structural basis of their high thermostability. *Journal of molecular biology*, 369(4), 1087–1097. https://doi.org/10.1016/j.jmb.2007.04.016

Faßhauer, P., Busche, T., Kalinowski, J., Mäder, U., Poehlein, A., Daniel, R., & Stülke, J. (2021). Functional Redundancy and Specialization of the Conserved Cold Shock Proteins in Bacillus subtilis. Microorganisms, 9(7), 1434. https://doi.org/10.3390/microorganisms9071434

Mueller, U., Perl, D., Schmid, F. X., & Heinemann, U. (2000). Thermal stability and atomic-resolution crystal structure of the Bacillus caldolyticus cold shock protein 1 1Edited by D. C. Rees. *Journal Of Molecular Biology*, 297(4), 975-988. https://doi.org/10.1006/jmbi.2000.3602

Perl, D., Mueller, U., Heinemann, U., & Schmid, F. X. (2000). Two exposed amino acid residues confer thermostability on a cold shock protein. Nature structural biology, 7(5), 380–383. https://doi.org/10.1038/75151

Schindelin, H., Marahiel, M. A., & Heinemann, U. (1993). Universal nucleic acid-binding domain revealed by crystal structure of the B. subtilis major cold-shock protein. Nature, 364(6433), 164–168. https://doi.org/10.1038/364164a0

Watanabe, K., & Suzuki, Y. (1998). Protein thermostabilization by proline substitutions. Journal Of Molecular Catalysis B Enzymatic, 4(4), 167-180. https://doi.org/10.1016/s1381-1177(97)00031-3

Wunderlich, M., Martin, A., & Schmid, F. X. (2005). Stabilization of the cold shock protein CspB from Bacillus subtilis by evolutionary optimization of Coulombic interactions. Journal of molecular biology, 347(5), 1063–1076. https://doi.org/10.1016/j.jmb.2005.02.014

